docs/design/mizar_arktos_scaleout.md

```diff
@@ -0,0 +1,170 @@
+Phu Tran                <@phudtran>         
+
+-->
+
+# Mizar Pod Networking Design for Arktos Scaleout Architecture
+
+## Introduction
+
+Scaleout architecture of Arktos re-imagines the way Kubernetes processes are deployed with the goal of scaling beyond what is possible with a single Kubernetes cluster.
+
+At a high level, this is achieved by deploying two different types Kubernetes clusters: resource partitions and tenant partitions.
+  - Resource partitions are specialized Kubernetes clusters where apiserver and node controller manage a set of Nodes.
+  - Tenant partitions are specialized Kubernetes clusters where apiserver, scheduler, and controllers manage a set of Pods for a particular tenant.
+
+Kubelets on each node in each resource partition connect to apiservers of each tenant partition and watch for Pods scheduled to the node. 
+Schedulers on each tenant partition connect to the apiservers of each resource partition and get a full view of all the nodes across all resource partitions, and the pods they host.
+
+This way of partitioning effectively distributes pod and node management tasks and their etcd object store requirements across a larger set of resources and allows scaling beyond what a single cluster can do.
+
+Scaleout architecture creates a very interesting problem for Pod Networking. Since each resource partition is effectively an 'isolated' Kubernetes cluster, we have two challenges:
+ 1. Available networking solutions that can be deployed in a cluster don't allow for communication of Pods belonging to a tenant across two or more resource partitions. 
+ 1. Available networking solutions don't provide network isolation between Pods belonging to different tenants in a resource partition. 
+
+Mizar was built with multitenancy in mind and offers the ability to create VPCs and deploy pods in those VPCs. By default, Pods in a VPC cannot communicate with Pods in another VPC.
+This natively provides strong network isolation for Pods in different VPCs without needing to rely on Kubernetes Network policies.
+
+This design document explores the various design options to deploy Mizar in Arktos Scaleout cluster that leverages VPCs for pod isolation on per-tenant basis. It then proposes a design that can handle the scale required by Arktos scaleout architecture.
+
+
+## Design Options for Arktos Scaleout Architecture
+
+1. Deploy Mizar in each tenant partition as it is deployed in a scaleup cluster.
+  - Pros:
+    - In theory, each tenant would have networking for its own Pods and this is natively isolated from other tenant Pods as long as Geneve VNI is unique.
+    - Pod information available from tenant-local apiserver, therefore little to no changes needed to the Mizar operator.
+  - Cons:
+    - Daemonset support needed in each tenant.
+    - Each Mizar deployment will want to start its own transit daemon. Significant work needed to support multiple transit daemons.
+
+2. Deploy Mizar in each resource partition.
```

Comment:
Current Arktos does not allow Mizar to be deployed in resource partition.

Reply:
I know. This section only mentions the options considered to inform about the different possible approaches we looked at and their pros and cons.